src/main/java/net/minecraftforge/common/extensions/IForgeResourceKey.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.common.extensions;
+
+import net.minecraft.core.Holder;
+import net.minecraft.core.RegistryAccess;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.entity.BlockEntity;
+
+import java.util.Objects;
+
+public interface IForgeResourceKey<T> {
+    private ResourceKey<T> self() {
+        return (ResourceKey<T>) this;
+    }
+
+    default Holder<T> getOrThrow(BlockEntity blockEntity) {
+        return getOrThrow(blockEntity.getLevel());
+    }
```

Comment:
I don't think this method is a good idea. You should really be directly calling `getOrThrow(Level)` if you want to do it like this.